miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java

```diff
@@ -177,7 +175,11 @@ protected PoolDto asDto(Pool model) {
   @ResponseBody
   public PoolDto createPool(@RequestBody PoolDto pool, UriComponentsBuilder uriBuilder, HttpServletResponse response)
       throws IOException {
-    Long id = poolService.save(Dtos.to(pool));
+    Pool poolobj = Dtos.to(pool);
+    if (poolobj.getVolume() == null && poolobj.getPoolableElementViews().stream().allMatch(view -> view.getDilutionVolumeUsed() != null)) {
```

Comment:
If the calculation is done here, then the user does not see what the projected pool volume is before save (and has to click the "Edit" button to see what it is after save). Most of our pools are created through the bulk pages, and we know at that point which dilutions are being used for the pool. Can this be moved to the EditLibraryController in the `poolBulk*Backend` methods?

Reply:
The only problem I can see with that would be when doing custom pooling. I'll see if I can get it to work as you suggested for the other forms of pooling and leave it as is for custom pooling.